<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Bioinformatics tools and pipelines using R and CWL</title>
  <meta name="description" content="A manual to wrap Bioinformatics tools and workflows with Common Workflow Language using the Rcwl package." />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Bioinformatics tools and pipelines using R and CWL" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A manual to wrap Bioinformatics tools and workflows with Common Workflow Language using the Rcwl package." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Bioinformatics tools and pipelines using R and CWL" />
  
  <meta name="twitter:description" content="A manual to wrap Bioinformatics tools and workflows with Common Workflow Language using the Rcwl package." />
  

<meta name="author" content="Qiang Hu" />
<meta name="author" content="Qian Liu" />


<meta name="date" content="2019-10-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="intro.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/d3-3.3.8/d3.min.js"></script>
<script src="libs/dagre-0.4.0/dagre-d3.min.js"></script>
<link href="libs/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet" />
<script src="libs/mermaid-0.3.0/dist/mermaid.slim.min.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/chromatography-0.1/chromatography.js"></script>
<script src="libs/DiagrammeR-binding-1.0.1/DiagrammeR.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rcwl and Case Study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#r-package-installation"><i class="fa fa-check"></i><b>0.1</b> R package installation</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#system-requirements"><i class="fa fa-check"></i><b>0.2</b> System requirements</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#docker"><i class="fa fa-check"></i><b>0.3</b> Docker</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#structure-of-the-book"><i class="fa fa-check"></i><b>0.4</b> Structure of the book</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i><b>0.5</b> R session information</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#common-workflow-language"><i class="fa fa-check"></i><b>1.1</b> Common Workflow Language</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#first-example"><i class="fa fa-check"></i><b>1.2</b> First example</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#test-run"><i class="fa fa-check"></i><b>1.3</b> Test run</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="components.html"><a href="components.html"><i class="fa fa-check"></i><b>2</b> Components</a><ul>
<li class="chapter" data-level="2.1" data-path="components.html"><a href="components.html#input-parameters"><i class="fa fa-check"></i><b>2.1</b> Input Parameters</a><ul>
<li class="chapter" data-level="2.1.1" data-path="components.html"><a href="components.html#essential-input-parameters"><i class="fa fa-check"></i><b>2.1.1</b> Essential Input parameters</a></li>
<li class="chapter" data-level="2.1.2" data-path="components.html"><a href="components.html#array-inputs"><i class="fa fa-check"></i><b>2.1.2</b> Array Inputs</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="components.html"><a href="components.html#output-parameters"><i class="fa fa-check"></i><b>2.2</b> Output Parameters</a><ul>
<li class="chapter" data-level="2.2.1" data-path="components.html"><a href="components.html#capturing-output"><i class="fa fa-check"></i><b>2.2.1</b> Capturing Output</a></li>
<li class="chapter" data-level="2.2.2" data-path="components.html"><a href="components.html#array-outputs"><i class="fa fa-check"></i><b>2.2.2</b> Array Outputs</a></li>
<li class="chapter" data-level="2.2.3" data-path="components.html"><a href="components.html#standard-output"><i class="fa fa-check"></i><b>2.2.3</b> Standard output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="writing-pipeline.html"><a href="writing-pipeline.html"><i class="fa fa-check"></i><b>3</b> Writing Pipeline</a><ul>
<li class="chapter" data-level="3.1" data-path="writing-pipeline.html"><a href="writing-pipeline.html#scattering-pipeline"><i class="fa fa-check"></i><b>3.1</b> Scattering pipeline</a></li>
<li class="chapter" data-level="3.2" data-path="writing-pipeline.html"><a href="writing-pipeline.html#pipeline-plot"><i class="fa fa-check"></i><b>3.2</b> Pipeline plot</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="run-approaches.html"><a href="run-approaches.html"><i class="fa fa-check"></i><b>4</b> Run approaches</a><ul>
<li class="chapter" data-level="4.1" data-path="run-approaches.html"><a href="run-approaches.html#running-tools-in-docker"><i class="fa fa-check"></i><b>4.1</b> Running Tools in Docker</a></li>
<li class="chapter" data-level="4.2" data-path="run-approaches.html"><a href="run-approaches.html#running-tools-in-cluster-server"><i class="fa fa-check"></i><b>4.2</b> Running Tools in Cluster server</a></li>
<li class="chapter" data-level="4.3" data-path="run-approaches.html"><a href="run-approaches.html#web-application"><i class="fa fa-check"></i><b>4.3</b> Web Application</a><ul>
<li class="chapter" data-level="4.3.1" data-path="run-approaches.html"><a href="run-approaches.html#cwlparam-example"><i class="fa fa-check"></i><b>4.3.1</b> cwlParam example</a></li>
<li class="chapter" data-level="4.3.2" data-path="run-approaches.html"><a href="run-approaches.html#cwlparam-to-shiny-app"><i class="fa fa-check"></i><b>4.3.2</b> cwlParam to Shiny App</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="application.html"><a href="application.html"><i class="fa fa-check"></i><b>5</b> Application</a><ul>
<li class="chapter" data-level="5.1" data-path="application.html"><a href="application.html#rcwlpipelines-tools"><i class="fa fa-check"></i><b>5.1</b> RcwlPipelines tools</a><ul>
<li class="chapter" data-level="5.1.1" data-path="application.html"><a href="application.html#rcwl-scripts"><i class="fa fa-check"></i><b>5.1.1</b> Rcwl scripts</a></li>
<li class="chapter" data-level="5.1.2" data-path="application.html"><a href="application.html#build-a-pipeline"><i class="fa fa-check"></i><b>5.1.2</b> Build a pipeline</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="application.html"><a href="application.html#rcwlpipelines-summary"><i class="fa fa-check"></i><b>5.2</b> RcwlPipelines summary</a></li>
<li class="chapter" data-level="5.3" data-path="application.html"><a href="application.html#dnaseq-alignment-pipeline"><i class="fa fa-check"></i><b>5.3</b> DNASeq alignment pipeline</a><ul>
<li class="chapter" data-level="5.3.1" data-path="application.html"><a href="application.html#prepare-data"><i class="fa fa-check"></i><b>5.3.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.3.2" data-path="application.html"><a href="application.html#run-in-cluster"><i class="fa fa-check"></i><b>5.3.2</b> Run in cluster</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="application.html"><a href="application.html#rnaseq-pipeline"><i class="fa fa-check"></i><b>5.4</b> RNASeq pipeline</a><ul>
<li class="chapter" data-level="5.4.1" data-path="application.html"><a href="application.html#prepare-data-1"><i class="fa fa-check"></i><b>5.4.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.4.2" data-path="application.html"><a href="application.html#submit-pipeline-with-sge"><i class="fa fa-check"></i><b>5.4.2</b> Submit pipeline with SGE</a></li>
<li class="chapter" data-level="5.4.3" data-path="application.html"><a href="application.html#summarize-qc"><i class="fa fa-check"></i><b>5.4.3</b> Summarize QC</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="application.html"><a href="application.html#mc3-somatic-variant-calling-pipeline"><i class="fa fa-check"></i><b>5.5</b> MC3 somatic variant calling pipeline</a><ul>
<li class="chapter" data-level="5.5.1" data-path="application.html"><a href="application.html#prepare-data-2"><i class="fa fa-check"></i><b>5.5.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.5.2" data-path="application.html"><a href="application.html#run-mc3-pipeline"><i class="fa fa-check"></i><b>5.5.2</b> Run MC3 pipeline</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="application.html"><a href="application.html#gatk4-germline-variant-calling-pipeline"><i class="fa fa-check"></i><b>5.6</b> GATK4 germline variant calling pipeline</a><ul>
<li class="chapter" data-level="5.6.1" data-path="application.html"><a href="application.html#gatk-alignment"><i class="fa fa-check"></i><b>5.6.1</b> GATK Alignment</a></li>
<li class="chapter" data-level="5.6.2" data-path="application.html"><a href="application.html#haplotypecaller"><i class="fa fa-check"></i><b>5.6.2</b> HaplotypeCaller</a></li>
<li class="chapter" data-level="5.6.3" data-path="application.html"><a href="application.html#joint-discovery"><i class="fa fa-check"></i><b>5.6.3</b> Joint Discovery</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html"><i class="fa fa-check"></i><b>6</b> DNA-Seq Variant calling</a><ul>
<li class="chapter" data-level="6.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#dna-alignment"><i class="fa fa-check"></i><b>6.1</b> DNA alignment</a></li>
<li class="chapter" data-level="6.2" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#germline-variant-calling"><i class="fa fa-check"></i><b>6.2</b> Germline variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#somatic-mutation-calling"><i class="fa fa-check"></i><b>6.3</b> Somatic mutation calling</a><ul>
<li class="chapter" data-level="6.3.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#mutect2"><i class="fa fa-check"></i><b>6.3.1</b> MuTect2</a></li>
<li class="chapter" data-level="6.3.2" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#muse"><i class="fa fa-check"></i><b>6.3.2</b> MuSE</a></li>
<li class="chapter" data-level="6.3.3" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#strelka2"><i class="fa fa-check"></i><b>6.3.3</b> Strelka2</a></li>
<li class="chapter" data-level="6.3.4" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#somaticsniper"><i class="fa fa-check"></i><b>6.3.4</b> SomaticSniper</a></li>
<li class="chapter" data-level="6.3.5" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#vardict"><i class="fa fa-check"></i><b>6.3.5</b> VarDict</a></li>
<li class="chapter" data-level="6.3.6" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#lofreq"><i class="fa fa-check"></i><b>6.3.6</b> LoFreq</a></li>
<li class="chapter" data-level="6.3.7" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#varscan2"><i class="fa fa-check"></i><b>6.3.7</b> VarScan2</a></li>
<li class="chapter" data-level="6.3.8" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#lancet"><i class="fa fa-check"></i><b>6.3.8</b> Lancet</a></li>
<li class="chapter" data-level="6.3.9" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#neusomatic"><i class="fa fa-check"></i><b>6.3.9</b> Neusomatic</a></li>
<li class="chapter" data-level="6.3.10" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#combined-somatic-caller-pipeline"><i class="fa fa-check"></i><b>6.3.10</b> Combined somatic caller pipeline</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#structural-variation"><i class="fa fa-check"></i><b>6.4</b> Structural variation</a><ul>
<li class="chapter" data-level="6.4.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#structural-variation-engine"><i class="fa fa-check"></i><b>6.4.1</b> Structural variation Engine</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#copy-number-variation"><i class="fa fa-check"></i><b>6.5</b> Copy number variation</a><ul>
<li class="chapter" data-level="6.5.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#cnvkit"><i class="fa fa-check"></i><b>6.5.1</b> cnvkit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html"><i class="fa fa-check"></i><b>7</b> RNA-Seq alignment and quantification</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#bulk-mrna"><i class="fa fa-check"></i><b>7.1</b> bulk mRNA</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#prepare-data-3"><i class="fa fa-check"></i><b>7.1.1</b> Prepare data</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#submit-pipeline-with-sge-1"><i class="fa fa-check"></i><b>7.1.2</b> Submit pipeline with SGE</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#qc-summary"><i class="fa fa-check"></i><b>7.1.3</b> QC Summary</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#abundances-summary"><i class="fa fa-check"></i><b>7.1.4</b> Abundances summary</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#transcriptome-quantification"><i class="fa fa-check"></i><b>7.2</b> transcriptome quantification</a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#kallisto"><i class="fa fa-check"></i><b>7.2.1</b> Kallisto</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#single-cell-rnaseq"><i class="fa fa-check"></i><b>7.3</b> single cell RNAseq</a></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#mirna"><i class="fa fa-check"></i><b>7.4</b> miRNA</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#prepare-reference"><i class="fa fa-check"></i><b>7.4.1</b> Prepare reference</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#run-mirdeep2-pipeline"><i class="fa fa-check"></i><b>7.4.2</b> Run miRDeep2 pipeline</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html"><i class="fa fa-check"></i><b>8</b> Neoantigen prediction</a><ul>
<li class="chapter" data-level="8.1" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#input-file-preparation"><i class="fa fa-check"></i><b>8.1</b> Input file preparation</a><ul>
<li class="chapter" data-level="8.1.1" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#annotating-vcf"><i class="fa fa-check"></i><b>8.1.1</b> Annotating VCF</a></li>
<li class="chapter" data-level="8.1.2" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#adding-coverage"><i class="fa fa-check"></i><b>8.1.2</b> Adding coverage</a></li>
<li class="chapter" data-level="8.1.3" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#adding-expression"><i class="fa fa-check"></i><b>8.1.3</b> Adding expression</a></li>
<li class="chapter" data-level="8.1.4" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#creating-a-phased-vcf"><i class="fa fa-check"></i><b>8.1.4</b> Creating a phased VCF</a></li>
<li class="chapter" data-level="8.1.5" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#combined-vcf-preparing-pipeline"><i class="fa fa-check"></i><b>8.1.5</b> Combined VCF preparing pipeline</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#hla-typing"><i class="fa fa-check"></i><b>8.2</b> HLA typing</a><ul>
<li class="chapter" data-level="8.2.1" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#polysolver"><i class="fa fa-check"></i><b>8.2.1</b> polysolver</a></li>
<li class="chapter" data-level="8.2.2" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#optitype"><i class="fa fa-check"></i><b>8.2.2</b> Optitype</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#neoantigen-prediction-by-pvacseq"><i class="fa fa-check"></i><b>8.3</b> Neoantigen prediction by pVACseq</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html"><i class="fa fa-check"></i><b>9</b> Other Sequencing pipelines</a><ul>
<li class="chapter" data-level="9.1" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html#chip-seq"><i class="fa fa-check"></i><b>9.1</b> ChiP-Seq</a></li>
<li class="chapter" data-level="9.2" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html#atac-seq"><i class="fa fa-check"></i><b>9.2</b> ATAC-Seq</a></li>
<li class="chapter" data-level="9.3" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html#microbiome-sequencing"><i class="fa fa-check"></i><b>9.3</b> Microbiome Sequencing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics tools and pipelines using R and CWL</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Bioinformatics tools and pipelines using R and CWL</h1>
<p class="author"><em>Qiang Hu</em></p>
<p class="author"><em>Qian Liu</em></p>
<p class="date"><em>2019-10-11</em></p>
</div>
<div id="preface" class="section level1 unnumbered">
<h1>Preface</h1>
<p>This short book introduces the R packages, <strong>Rcwl</strong> and <strong>RcwlPipelines</strong>, to improve the way of building, managing and running Bioinformatics tools and pipelines.</p>
<p>The <strong>Rcwl</strong> package is built on top of the Common Workflow Language (CWL), and provides a simple and user-friendly way to wrap command line tools into data analysis pipelines in R. The <strong>RcwlPipelines</strong> package is a collection of Bioinformatics tools and pipelines based on <strong>Rcwl</strong>.</p>
<div id="r-package-installation" class="section level2">
<h2><span class="header-section-number">0.1</span> R package installation</h2>
<p>The <strong>Rcwl</strong> and <strong>RcwlPipelines</strong> packages can be installed from Bioconductor or Github:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="kw">c</span>(<span class="st">&quot;Rcwl&quot;</span>, <span class="st">&quot;RcwlPipelines&quot;</span>))</span>
<span id="cb1-2"><a href="index.html#cb1-2"></a><span class="co"># or the development version</span></span>
<span id="cb1-3"><a href="index.html#cb1-3"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="kw">c</span>(<span class="st">&quot;hubentu/Rcwl&quot;</span>, <span class="st">&quot;hubentu/RcwlPipelines&quot;</span>))</span></code></pre></div>
<p>To load the packages into R session:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1"></a><span class="kw">library</span>(Rcwl)</span>
<span id="cb2-2"><a href="index.html#cb2-2"></a><span class="kw">library</span>(RcwlPipelines)</span></code></pre></div>
</div>
<div id="system-requirements" class="section level2">
<h2><span class="header-section-number">0.2</span> System requirements</h2>
<p>In addition to the R packages, the following tools are required to be installed to run the examples in this book.</p>
<ul>
<li>python (&gt;= 2.7)</li>
<li>cwltool (&gt;= 1.0.2018)</li>
<li>nodejs</li>
</ul>
<p>The <strong>cwltool</strong> is the reference implementation of the Common Workflow Language, which is used to run the CWL scripts. The <strong>nodejs</strong> is required when the CWL scripts use JavaScript. You can find instructions to install these tools here:</p>
<ul>
<li><a href="https://github.com/common-workflow-language/cwltool#install" class="uri">https://github.com/common-workflow-language/cwltool#install</a></li>
<li><a href="https://nodejs.org" class="uri">https://nodejs.org</a></li>
</ul>
</div>
<div id="docker" class="section level2">
<h2><span class="header-section-number">0.3</span> Docker</h2>
<p>The Docker container simplifies software installation and management, especially for bioinformatics tools/pipelines requiring different runtime environments and library dependencies. A CWL runner can perform this work automatically by pulling the Docker containers and mounting the paths of input files.</p>
<p>The Docker requirement is optional, as CWL scripts can also be run locally with all the dependencies pre-installed.</p>
</div>
<div id="structure-of-the-book" class="section level2">
<h2><span class="header-section-number">0.4</span> Structure of the book</h2>
<ul>
<li>Introduction</li>
<li>Components</li>
<li>Build CWL workflows</li>
<li>Run approaches</li>
<li>Case study</li>
</ul>
</div>
<div id="r-session-information" class="section level2">
<h2><span class="header-section-number">0.5</span> R session information</h2>
<p>The R session information for compiling this mannual is shown below:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /home/qhu/.linuxbrew/Cellar/openblas/0.3.7/lib/libopenblasp-r0.3.7.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bookdown_0.14        RcwlPipelines_1.1.12 testthat_2.2.1      
##  [4] BiocFileCache_1.8.0  dbplyr_1.4.2         Rcwl_1.1.13         
##  [7] S4Vectors_0.22.1     BiocGenerics_0.30.0  yaml_2.2.0          
## [10] devtools_2.2.1       usethis_1.5.1       
## 
## loaded via a namespace (and not attached):
##  [1] fs_1.3.1            bit64_0.9-7         httr_1.4.1         
##  [4] RColorBrewer_1.1-2  progress_1.2.2      rprojroot_1.3-2    
##  [7] tools_3.6.1         backports_1.1.4     R6_2.4.0           
## [10] DBI_1.0.0           lazyeval_0.2.2      colorspace_1.4-1   
## [13] withr_2.1.2         tidyselect_0.2.5    gridExtra_2.3      
## [16] prettyunits_1.0.2   processx_3.4.1      curl_4.2           
## [19] bit_1.1-14          compiler_3.6.1      cli_1.1.0          
## [22] influenceR_0.1.0    desc_1.2.0          scales_1.0.0       
## [25] checkmate_1.9.4     readr_1.3.1         callr_3.3.2        
## [28] rappdirs_0.3.1      stringr_1.4.0       digest_0.6.21      
## [31] rmarkdown_1.16      R.utils_2.9.0       pkgconfig_2.0.3    
## [34] htmltools_0.3.6     sessioninfo_1.1.1   htmlwidgets_1.3    
## [37] rlang_0.4.0         rstudioapi_0.10     RSQLite_2.1.2      
## [40] shiny_1.3.2         visNetwork_2.0.8    jsonlite_1.6       
## [43] BiocParallel_1.18.1 dplyr_0.8.3         rgexf_0.15.3       
## [46] R.oo_1.22.0         magrittr_1.5        Rcpp_1.0.2         
## [49] munsell_0.5.0       viridis_0.5.1       lifecycle_0.1.0    
## [52] R.methodsS3_1.7.1   stringi_1.4.3       pkgbuild_1.0.5     
## [55] blob_1.2.0          grid_3.6.1          promises_1.0.1     
## [58] crayon_1.3.4        hms_0.5.1           batchtools_0.9.11  
## [61] knitr_1.25          zeallot_0.1.0       ps_1.3.0           
## [64] pillar_1.4.2        igraph_1.2.4.1      base64url_1.4      
## [67] codetools_0.2-16    pkgload_1.0.2       XML_3.98-1.20      
## [70] glue_1.3.1          evaluate_0.14       downloader_0.4     
## [73] data.table_1.12.2   remotes_2.1.0       vctrs_0.2.0        
## [76] httpuv_1.5.2        gtable_0.3.0        purrr_0.3.2        
## [79] tidyr_1.0.0         assertthat_0.2.1    ggplot2_3.2.1      
## [82] xfun_0.10           mime_0.7            xtable_1.8-4       
## [85] later_0.8.0         viridisLite_0.3.0   tibble_2.1.3       
## [88] memoise_1.1.0       Rook_1.1-1          DiagrammeR_1.0.1   
## [91] ellipsis_0.3.0      brew_1.0-6</code></pre>


</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="intro.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["Rcwl-Pipelines.pdf", "Rcwl-Pipelines.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
