<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 RNA-Seq alignment and quantification | Bioinformatics tools and pipelines using R and CWL</title>
  <meta name="description" content="A manual to wrap Bioinformatics tools and workflows with Common Workflow Language using the Rcwl package." />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 RNA-Seq alignment and quantification | Bioinformatics tools and pipelines using R and CWL" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A manual to wrap Bioinformatics tools and workflows with Common Workflow Language using the Rcwl package." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 RNA-Seq alignment and quantification | Bioinformatics tools and pipelines using R and CWL" />
  
  <meta name="twitter:description" content="A manual to wrap Bioinformatics tools and workflows with Common Workflow Language using the Rcwl package." />
  

<meta name="author" content="Qiang Hu" />
<meta name="author" content="Qian Liu" />


<meta name="date" content="2019-10-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dna-seq-variant-calling.html"/>
<link rel="next" href="neoantigen-prediction.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/d3-3.3.8/d3.min.js"></script>
<script src="libs/dagre-0.4.0/dagre-d3.min.js"></script>
<link href="libs/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet" />
<script src="libs/mermaid-0.3.0/dist/mermaid.slim.min.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/chromatography-0.1/chromatography.js"></script>
<script src="libs/DiagrammeR-binding-1.0.1/DiagrammeR.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Rcwl and Case Study</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#r-package-installation"><i class="fa fa-check"></i><b>0.1</b> R package installation</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#system-requirements"><i class="fa fa-check"></i><b>0.2</b> System requirements</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#docker"><i class="fa fa-check"></i><b>0.3</b> Docker</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#structure-of-the-book"><i class="fa fa-check"></i><b>0.4</b> Structure of the book</a></li>
<li class="chapter" data-level="0.5" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i><b>0.5</b> R session information</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#common-workflow-language"><i class="fa fa-check"></i><b>1.1</b> Common Workflow Language</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#first-example"><i class="fa fa-check"></i><b>1.2</b> First example</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#test-run"><i class="fa fa-check"></i><b>1.3</b> Test run</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="components.html"><a href="components.html"><i class="fa fa-check"></i><b>2</b> Components</a><ul>
<li class="chapter" data-level="2.1" data-path="components.html"><a href="components.html#input-parameters"><i class="fa fa-check"></i><b>2.1</b> Input Parameters</a><ul>
<li class="chapter" data-level="2.1.1" data-path="components.html"><a href="components.html#essential-input-parameters"><i class="fa fa-check"></i><b>2.1.1</b> Essential Input parameters</a></li>
<li class="chapter" data-level="2.1.2" data-path="components.html"><a href="components.html#array-inputs"><i class="fa fa-check"></i><b>2.1.2</b> Array Inputs</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="components.html"><a href="components.html#output-parameters"><i class="fa fa-check"></i><b>2.2</b> Output Parameters</a><ul>
<li class="chapter" data-level="2.2.1" data-path="components.html"><a href="components.html#capturing-output"><i class="fa fa-check"></i><b>2.2.1</b> Capturing Output</a></li>
<li class="chapter" data-level="2.2.2" data-path="components.html"><a href="components.html#array-outputs"><i class="fa fa-check"></i><b>2.2.2</b> Array Outputs</a></li>
<li class="chapter" data-level="2.2.3" data-path="components.html"><a href="components.html#standard-output"><i class="fa fa-check"></i><b>2.2.3</b> Standard output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="writing-pipeline.html"><a href="writing-pipeline.html"><i class="fa fa-check"></i><b>3</b> Writing Pipeline</a><ul>
<li class="chapter" data-level="3.1" data-path="writing-pipeline.html"><a href="writing-pipeline.html#scattering-pipeline"><i class="fa fa-check"></i><b>3.1</b> Scattering pipeline</a></li>
<li class="chapter" data-level="3.2" data-path="writing-pipeline.html"><a href="writing-pipeline.html#pipeline-plot"><i class="fa fa-check"></i><b>3.2</b> Pipeline plot</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="run-approaches.html"><a href="run-approaches.html"><i class="fa fa-check"></i><b>4</b> Run approaches</a><ul>
<li class="chapter" data-level="4.1" data-path="run-approaches.html"><a href="run-approaches.html#running-tools-in-docker"><i class="fa fa-check"></i><b>4.1</b> Running Tools in Docker</a></li>
<li class="chapter" data-level="4.2" data-path="run-approaches.html"><a href="run-approaches.html#running-tools-in-cluster-server"><i class="fa fa-check"></i><b>4.2</b> Running Tools in Cluster server</a></li>
<li class="chapter" data-level="4.3" data-path="run-approaches.html"><a href="run-approaches.html#web-application"><i class="fa fa-check"></i><b>4.3</b> Web Application</a><ul>
<li class="chapter" data-level="4.3.1" data-path="run-approaches.html"><a href="run-approaches.html#cwlparam-example"><i class="fa fa-check"></i><b>4.3.1</b> cwlParam example</a></li>
<li class="chapter" data-level="4.3.2" data-path="run-approaches.html"><a href="run-approaches.html#cwlparam-to-shiny-app"><i class="fa fa-check"></i><b>4.3.2</b> cwlParam to Shiny App</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="application.html"><a href="application.html"><i class="fa fa-check"></i><b>5</b> Application</a><ul>
<li class="chapter" data-level="5.1" data-path="application.html"><a href="application.html#rcwlpipelines-tools"><i class="fa fa-check"></i><b>5.1</b> RcwlPipelines tools</a><ul>
<li class="chapter" data-level="5.1.1" data-path="application.html"><a href="application.html#rcwl-scripts"><i class="fa fa-check"></i><b>5.1.1</b> Rcwl scripts</a></li>
<li class="chapter" data-level="5.1.2" data-path="application.html"><a href="application.html#build-a-pipeline"><i class="fa fa-check"></i><b>5.1.2</b> Build a pipeline</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="application.html"><a href="application.html#rcwlpipelines-summary"><i class="fa fa-check"></i><b>5.2</b> RcwlPipelines summary</a></li>
<li class="chapter" data-level="5.3" data-path="application.html"><a href="application.html#dnaseq-alignment-pipeline"><i class="fa fa-check"></i><b>5.3</b> DNASeq alignment pipeline</a><ul>
<li class="chapter" data-level="5.3.1" data-path="application.html"><a href="application.html#prepare-data"><i class="fa fa-check"></i><b>5.3.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.3.2" data-path="application.html"><a href="application.html#run-in-cluster"><i class="fa fa-check"></i><b>5.3.2</b> Run in cluster</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="application.html"><a href="application.html#rnaseq-pipeline"><i class="fa fa-check"></i><b>5.4</b> RNASeq pipeline</a><ul>
<li class="chapter" data-level="5.4.1" data-path="application.html"><a href="application.html#prepare-data-1"><i class="fa fa-check"></i><b>5.4.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.4.2" data-path="application.html"><a href="application.html#submit-pipeline-with-sge"><i class="fa fa-check"></i><b>5.4.2</b> Submit pipeline with SGE</a></li>
<li class="chapter" data-level="5.4.3" data-path="application.html"><a href="application.html#summarize-qc"><i class="fa fa-check"></i><b>5.4.3</b> Summarize QC</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="application.html"><a href="application.html#mc3-somatic-variant-calling-pipeline"><i class="fa fa-check"></i><b>5.5</b> MC3 somatic variant calling pipeline</a><ul>
<li class="chapter" data-level="5.5.1" data-path="application.html"><a href="application.html#prepare-data-2"><i class="fa fa-check"></i><b>5.5.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.5.2" data-path="application.html"><a href="application.html#run-mc3-pipeline"><i class="fa fa-check"></i><b>5.5.2</b> Run MC3 pipeline</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="application.html"><a href="application.html#gatk4-germline-variant-calling-pipeline"><i class="fa fa-check"></i><b>5.6</b> GATK4 germline variant calling pipeline</a><ul>
<li class="chapter" data-level="5.6.1" data-path="application.html"><a href="application.html#gatk-alignment"><i class="fa fa-check"></i><b>5.6.1</b> GATK Alignment</a></li>
<li class="chapter" data-level="5.6.2" data-path="application.html"><a href="application.html#haplotypecaller"><i class="fa fa-check"></i><b>5.6.2</b> HaplotypeCaller</a></li>
<li class="chapter" data-level="5.6.3" data-path="application.html"><a href="application.html#joint-discovery"><i class="fa fa-check"></i><b>5.6.3</b> Joint Discovery</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html"><i class="fa fa-check"></i><b>6</b> DNA-Seq Variant calling</a><ul>
<li class="chapter" data-level="6.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#dna-alignment"><i class="fa fa-check"></i><b>6.1</b> DNA alignment</a></li>
<li class="chapter" data-level="6.2" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#germline-variant-calling"><i class="fa fa-check"></i><b>6.2</b> Germline variant calling</a></li>
<li class="chapter" data-level="6.3" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#somatic-mutation-calling"><i class="fa fa-check"></i><b>6.3</b> Somatic mutation calling</a><ul>
<li class="chapter" data-level="6.3.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#mutect2"><i class="fa fa-check"></i><b>6.3.1</b> MuTect2</a></li>
<li class="chapter" data-level="6.3.2" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#muse"><i class="fa fa-check"></i><b>6.3.2</b> MuSE</a></li>
<li class="chapter" data-level="6.3.3" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#strelka2"><i class="fa fa-check"></i><b>6.3.3</b> Strelka2</a></li>
<li class="chapter" data-level="6.3.4" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#somaticsniper"><i class="fa fa-check"></i><b>6.3.4</b> SomaticSniper</a></li>
<li class="chapter" data-level="6.3.5" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#vardict"><i class="fa fa-check"></i><b>6.3.5</b> VarDict</a></li>
<li class="chapter" data-level="6.3.6" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#lofreq"><i class="fa fa-check"></i><b>6.3.6</b> LoFreq</a></li>
<li class="chapter" data-level="6.3.7" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#varscan2"><i class="fa fa-check"></i><b>6.3.7</b> VarScan2</a></li>
<li class="chapter" data-level="6.3.8" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#lancet"><i class="fa fa-check"></i><b>6.3.8</b> Lancet</a></li>
<li class="chapter" data-level="6.3.9" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#neusomatic"><i class="fa fa-check"></i><b>6.3.9</b> Neusomatic</a></li>
<li class="chapter" data-level="6.3.10" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#combined-somatic-caller-pipeline"><i class="fa fa-check"></i><b>6.3.10</b> Combined somatic caller pipeline</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#structural-variation"><i class="fa fa-check"></i><b>6.4</b> Structural variation</a><ul>
<li class="chapter" data-level="6.4.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#structural-variation-engine"><i class="fa fa-check"></i><b>6.4.1</b> Structural variation Engine</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#copy-number-variation"><i class="fa fa-check"></i><b>6.5</b> Copy number variation</a><ul>
<li class="chapter" data-level="6.5.1" data-path="dna-seq-variant-calling.html"><a href="dna-seq-variant-calling.html#cnvkit"><i class="fa fa-check"></i><b>6.5.1</b> cnvkit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html"><i class="fa fa-check"></i><b>7</b> RNA-Seq alignment and quantification</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#bulk-mrna"><i class="fa fa-check"></i><b>7.1</b> bulk mRNA</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#prepare-data-3"><i class="fa fa-check"></i><b>7.1.1</b> Prepare data</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#submit-pipeline-with-sge-1"><i class="fa fa-check"></i><b>7.1.2</b> Submit pipeline with SGE</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#qc-summary"><i class="fa fa-check"></i><b>7.1.3</b> QC Summary</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#abundances-summary"><i class="fa fa-check"></i><b>7.1.4</b> Abundances summary</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#transcriptome-quantification"><i class="fa fa-check"></i><b>7.2</b> transcriptome quantification</a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#kallisto"><i class="fa fa-check"></i><b>7.2.1</b> Kallisto</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#single-cell-rnaseq"><i class="fa fa-check"></i><b>7.3</b> single cell RNAseq</a></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#mirna"><i class="fa fa-check"></i><b>7.4</b> miRNA</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#prepare-reference"><i class="fa fa-check"></i><b>7.4.1</b> Prepare reference</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-alignment-and-quantification.html"><a href="rna-seq-alignment-and-quantification.html#run-mirdeep2-pipeline"><i class="fa fa-check"></i><b>7.4.2</b> Run miRDeep2 pipeline</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html"><i class="fa fa-check"></i><b>8</b> Neoantigen prediction</a><ul>
<li class="chapter" data-level="8.1" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#input-file-preparation"><i class="fa fa-check"></i><b>8.1</b> Input file preparation</a><ul>
<li class="chapter" data-level="8.1.1" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#annotating-vcf"><i class="fa fa-check"></i><b>8.1.1</b> Annotating VCF</a></li>
<li class="chapter" data-level="8.1.2" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#adding-coverage"><i class="fa fa-check"></i><b>8.1.2</b> Adding coverage</a></li>
<li class="chapter" data-level="8.1.3" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#adding-expression"><i class="fa fa-check"></i><b>8.1.3</b> Adding expression</a></li>
<li class="chapter" data-level="8.1.4" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#creating-a-phased-vcf"><i class="fa fa-check"></i><b>8.1.4</b> Creating a phased VCF</a></li>
<li class="chapter" data-level="8.1.5" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#combined-vcf-preparing-pipeline"><i class="fa fa-check"></i><b>8.1.5</b> Combined VCF preparing pipeline</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#hla-typing"><i class="fa fa-check"></i><b>8.2</b> HLA typing</a><ul>
<li class="chapter" data-level="8.2.1" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#polysolver"><i class="fa fa-check"></i><b>8.2.1</b> polysolver</a></li>
<li class="chapter" data-level="8.2.2" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#optitype"><i class="fa fa-check"></i><b>8.2.2</b> Optitype</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="neoantigen-prediction.html"><a href="neoantigen-prediction.html#neoantigen-prediction-by-pvacseq"><i class="fa fa-check"></i><b>8.3</b> Neoantigen prediction by pVACseq</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html"><i class="fa fa-check"></i><b>9</b> Other Sequencing pipelines</a><ul>
<li class="chapter" data-level="9.1" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html#chip-seq"><i class="fa fa-check"></i><b>9.1</b> ChiP-Seq</a></li>
<li class="chapter" data-level="9.2" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html#atac-seq"><i class="fa fa-check"></i><b>9.2</b> ATAC-Seq</a></li>
<li class="chapter" data-level="9.3" data-path="other-sequencing-pipelines.html"><a href="other-sequencing-pipelines.html#microbiome-sequencing"><i class="fa fa-check"></i><b>9.3</b> Microbiome Sequencing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics tools and pipelines using R and CWL</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-seq-alignment-and-quantification" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> RNA-Seq alignment and quantification</h1>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="rna-seq-alignment-and-quantification.html#cb227-1"></a><span class="kw">library</span>(RcwlPipelines)</span>
<span id="cb227-2"><a href="rna-seq-alignment-and-quantification.html#cb227-2"></a><span class="kw">library</span>(BiocParallel)</span>
<span id="cb227-3"><a href="rna-seq-alignment-and-quantification.html#cb227-3"></a><span class="kw">library</span>(fs)</span>
<span id="cb227-4"><a href="rna-seq-alignment-and-quantification.html#cb227-4"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb227-5"><a href="rna-seq-alignment-and-quantification.html#cb227-5"></a><span class="kw">library</span>(conflicted)</span>
<span id="cb227-6"><a href="rna-seq-alignment-and-quantification.html#cb227-6"></a><span class="co">## always use select from dplyr</span></span>
<span id="cb227-7"><a href="rna-seq-alignment-and-quantification.html#cb227-7"></a><span class="kw">conflict_prefer</span>(<span class="st">&quot;select&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="dt">quiet =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div id="bulk-mrna" class="section level2">
<h2><span class="header-section-number">7.1</span> bulk mRNA</h2>
<p>The <code>rnaseq_Sf</code> pipeline was built with reads quality summary, <code>STAR</code> alignment,
quantification by <code>featureCounts</code> and <code>RSeQC</code> quality control. Here
are the inputs:</p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="rna-seq-alignment-and-quantification.html#cb228-1"></a><span class="kw">inputs</span>(rnaseq_Sf)</span></code></pre></div>
<pre><code>## inputs:
##   in_seqfiles (File[]):  
##   in_prefix (string):  
##   in_genomeDir (Directory):  
##   in_GTFfile (File):  
##   in_runThreadN (int):  1</code></pre>
<p>The pipeline includes 6 steps:</p>
<ul>
<li><code>fastqc</code>: To run quality summary for raw fastqs with <code>fastqc</code>.</li>
<li><code>STAR</code>: To align fastqs with <code>STAR</code>.</li>
<li><code>samtools_index</code>: To index aligned bam file.</li>
<li><code>samtools_flagstat</code>: To summarize alignment flags.</li>
<li><code>featureCounts</code>: To quantify gene abundances.</li>
<li><code>RSeQC</code>: Several steps included.<br />

<ul>
<li><code>gtfToGenePred</code>: To convert GTF annotation to “genePred” format.</li>
<li><code>genePredToBed</code>: To convert “genePred” annotation to “bed” format.</li>
<li><code>r_distribution</code>: To summarize reads distribution over genome features.</li>
<li><code>gCoverage</code>: To summarize read coverage over gene body.</li>
</ul></li>
</ul>
<p>The outputs and logs from alignment, quantification and QC steps are
collected together into the output folder. A final QC report could be
generated by <code>multiqc</code>, which is also available in the data package.</p>
<p>Here are the short summary and steps plot:</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="rna-seq-alignment-and-quantification.html#cb230-1"></a><span class="kw">short</span>(rnaseq_Sf)</span></code></pre></div>
<pre><code>## inputs:
## - in_seqfiles
## - in_prefix
## - in_genomeDir
## - in_GTFfile
## - in_runThreadN
## outputs:
## - out_fastqc
## - out_BAM
## - out_Log
## - out_Count
## - out_stat
## - out_count
## - out_distribution
## - out_gCovP
## - out_gCovT
## steps:
## - fastqc
## - STAR
## - sortBam
## - samtools_index
## - samtools_flagstat
## - featureCounts
## - gtfToGenePred
## - genePredToBed
## - r_distribution
## - gCoverage</code></pre>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="rna-seq-alignment-and-quantification.html#cb232-1"></a><span class="kw">plotCWL</span>(rnaseq_Sf)</span></code></pre></div>
<div id="htmlwidget-d053de577e81874a5d39" style="width:672px;height:480px;" class="DiagrammeR html-widget"></div>
<script type="application/json" data-for="htmlwidget-d053de577e81874a5d39">{"x":{"diagram":"graph TB\nA(in_seqfiles)-->O{fastqc}\nO{fastqc}-->F((out_fastqc))\nB(in_prefix)-->P{STAR}\nC(in_genomeDir)-->P{STAR}\nD(in_GTFfile)-->P{STAR}\nA(in_seqfiles)-->P{STAR}\nE(in_runThreadN)-->P{STAR}\nP{STAR}-->H((out_Log))\nP{STAR}-->I((out_Count))\nP{STAR}-->|outBAM|Q{sortBam}\nQ{sortBam}-->|sbam|R{samtools_index}\nR{samtools_index}-->G((out_BAM))\nQ{sortBam}-->|sbam|S{samtools_flagstat}\nS{samtools_flagstat}-->J((out_stat))\nD(in_GTFfile)-->T{featureCounts}\nG((out_BAM))-->T{featureCounts}\nT{featureCounts}-->K((out_count))\nD(in_GTFfile)-->U{gtfToGenePred}\nU{gtfToGenePred}-->|genePred|V{genePredToBed}\nG((out_BAM))-->W{r_distribution}\nV{genePredToBed}-->|bed|W{r_distribution}\nW{r_distribution}-->L((out_distribution))\nG((out_BAM))-->X{gCoverage}\nV{genePredToBed}-->|bed|X{gCoverage}\nX{gCoverage}-->M((out_gCovP))\nX{gCoverage}-->N((out_gCovT))"},"evals":[],"jsHooks":[]}</script>
<div id="prepare-data-3" class="section level3">
<h3><span class="header-section-number">7.1.1</span> Prepare data</h3>
<p>An RNASeq test data set can be downloaded from <a href="http://genomedata.org/rnaseq-tutorial/HBR_UHR_ERCC_ds_5pc.tar">genomedata</a>, which includes paired-end fastqs for 6 samples.</p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="rna-seq-alignment-and-quantification.html#cb233-1"></a><span class="kw">download.file</span>(<span class="st">&quot;http://genomedata.org/rnaseq-tutorial/HBR_UHR_ERCC_ds_5pc.tar&quot;</span>, <span class="st">&quot;apps/RNASeq/data/HBR_UHR_ERCC_ds_5pc.tar)</span></span>
<span id="cb233-2"><a href="rna-seq-alignment-and-quantification.html#cb233-2"></a><span class="st">untar(&quot;</span>apps<span class="op">/</span>RNASeq<span class="op">/</span>data<span class="op">/</span>HBR_UHR_ERCC_ds_5pc.tar<span class="st">&quot;, exdir = &quot;</span>apps<span class="op">/</span>RNASeq<span class="op">/</span>data<span class="op">/</span><span class="st">&quot;)</span></span></code></pre></div>
<p>The input data must be in a named list, with the same names as the input list of the pipeline. For this pipeline, 5 inputs are required to be set, including <code>in_seqfiles</code>, <code>in_prefix</code>, <code>in_genomeDir</code>, <code>in_GTFfile</code> and <code>in_runThreadN</code>.</p>
<p>There are two different input lists, <code>inputList</code> and <code>paramList</code>. The <code>inputList</code> is used to define the inputs for each sample and will be submitted to different cluster nodes. The <code>paramList</code> is used to define parameters which are shared in all jobs.</p>
<p>Two following inputs should be listed in <code>inputList</code>:</p>
<ul>
<li><p><code>in_seqfiles</code>: A list with the fastq files of each sample in each element. The names of the list are also required to be defined and can be the sample IDs. The length of the list will be the same as the number of samples, thus the list will be defined to <code>inputList</code> and assigned to different nodes for parallel computing.</p></li>
<li><p><code>in_prefix</code> is the same as <code>in_seqfiles</code>, which defines a list of sample IDs.</p></li>
</ul>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="rna-seq-alignment-and-quantification.html#cb234-1"></a>files &lt;-<span class="st"> </span><span class="kw">normalizePath</span>(<span class="kw">list.files</span>(<span class="st">&quot;apps/RNASeq/data/&quot;</span>, <span class="st">&quot;.gz&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>))</span>
<span id="cb234-2"><a href="rna-seq-alignment-and-quantification.html#cb234-2"></a>files &lt;-<span class="st"> </span><span class="kw">tapply</span>(files, <span class="kw">substring</span>(<span class="kw">basename</span>(files), <span class="dv">1</span>, <span class="dv">8</span>), as.list)</span>
<span id="cb234-3"><a href="rna-seq-alignment-and-quantification.html#cb234-3"></a>inputList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">in_seqfiles =</span> files,</span>
<span id="cb234-4"><a href="rna-seq-alignment-and-quantification.html#cb234-4"></a>                  <span class="dt">in_prefix =</span> <span class="kw">as.list</span>(<span class="kw">names</span>(files)))</span></code></pre></div>
<p>These 3 parameter will be defined in <code>paramList</code>:</p>
<ul>
<li><p><code>in_genomeDir</code>: The reference genome indexes for STAR.</p></li>
<li><p><code>in_GTFfile</code>: The gene annotation file in GTF format.</p></li>
<li><p><code>in_runThreadN</code>: The number of threads to run for each job.</p></li>
</ul>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="rna-seq-alignment-and-quantification.html#cb235-1"></a>paramList &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb235-2"><a href="rna-seq-alignment-and-quantification.html#cb235-2"></a>    <span class="dt">in_genomeDir =</span> <span class="st">&quot;apps/data/GRCh38_100/&quot;</span>,</span>
<span id="cb235-3"><a href="rna-seq-alignment-and-quantification.html#cb235-3"></a>    <span class="dt">in_GTFfile =</span> <span class="st">&quot;apps/data/gencode.v25.annotation.gtf&quot;</span>,</span>
<span id="cb235-4"><a href="rna-seq-alignment-and-quantification.html#cb235-4"></a>    <span class="dt">in_runThreadN =</span> <span class="dv">4</span></span>
<span id="cb235-5"><a href="rna-seq-alignment-and-quantification.html#cb235-5"></a>)</span></code></pre></div>
<p>In some cases, we need to modify the default arguments in some steps of a pipeline. For example,</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="rna-seq-alignment-and-quantification.html#cb236-1"></a><span class="kw">arguments</span>(rnaseq_Sf, <span class="st">&quot;STAR&quot;</span>)[[<span class="dv">2</span>]] &lt;-<span class="st"> &quot;2&quot;</span></span>
<span id="cb236-2"><a href="rna-seq-alignment-and-quantification.html#cb236-2"></a><span class="kw">head</span>(<span class="kw">arguments</span>(rnaseq_Sf, <span class="st">&quot;STAR&quot;</span>))</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;--outFilterMultimapNmax&quot;
## 
## [[2]]
## [1] &quot;2&quot;
## 
## [[3]]
## [1] &quot;--outSAMunmapped&quot;
## 
## [[4]]
## [1] &quot;Within&quot;
## 
## [[5]]
## [1] &quot;--outFilterMismatchNmax&quot;
## 
## [[6]]
## [1] &quot;2&quot;</code></pre>
</div>
<div id="submit-pipeline-with-sge-1" class="section level3">
<h3><span class="header-section-number">7.1.2</span> Submit pipeline with SGE</h3>
<p>The function <code>runCWLBatch</code> is used to submit the pipeline to cluster server. In addition to defining <code>inputList</code> and <code>paramList</code>, we need to define parallel parameters from the <code>BiocParallel</code> package. Here is an example where we use “sge” to submit the jobs. The “sge” template is a bash script with some predefined parameters for “qsub”. The nodes queue name and number of slots/threads are variables from the template and can be assigned by the <code>resources</code> list.</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="rna-seq-alignment-and-quantification.html#cb238-1"></a>res &lt;-<span class="st"> </span><span class="kw">runCWLBatch</span>(<span class="dt">cwl =</span> rnaseq_Sf, <span class="dt">outdir =</span> <span class="st">&quot;apps/RNASeq/output/&quot;</span>,</span>
<span id="cb238-2"><a href="rna-seq-alignment-and-quantification.html#cb238-2"></a>                   <span class="dt">inputList =</span> inputList, <span class="dt">paramList =</span> paramList,</span>
<span id="cb238-3"><a href="rna-seq-alignment-and-quantification.html#cb238-3"></a>                   <span class="dt">BPPARAM =</span> <span class="kw">BatchtoolsParam</span>(</span>
<span id="cb238-4"><a href="rna-seq-alignment-and-quantification.html#cb238-4"></a>                       <span class="dt">workers =</span> <span class="kw">lengths</span>(inputList)[<span class="dv">1</span>], <span class="dt">cluster =</span> <span class="st">&quot;sge&quot;</span>,</span>
<span id="cb238-5"><a href="rna-seq-alignment-and-quantification.html#cb238-5"></a>                       <span class="dt">template =</span> <span class="st">&quot;apps/data/sge_centos7.tmpl&quot;</span>,</span>
<span id="cb238-6"><a href="rna-seq-alignment-and-quantification.html#cb238-6"></a>                       <span class="dt">resources =</span> <span class="kw">list</span>(<span class="dt">queue =</span> <span class="st">&quot;centos7.q&quot;</span>,</span>
<span id="cb238-7"><a href="rna-seq-alignment-and-quantification.html#cb238-7"></a>                                        <span class="dt">threads =</span> <span class="dv">4</span>)))</span></code></pre></div>
<p>That’s it! The fastqc files of each sample will be submitted to different nodes to run the whole pipeline automatically.</p>
<p>All the results have been collected to output directory of each sample. For example,</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="rna-seq-alignment-and-quantification.html#cb239-1"></a><span class="kw">dir</span>(<span class="st">&quot;apps/RNASeq/output/HBR_Rep1&quot;</span>)</span></code></pre></div>
<pre><code>##  [1] &quot;abundance.h5&quot;                                                     
##  [2] &quot;abundance.tsv&quot;                                                    
##  [3] &quot;HBR_Rep1_abundance.tsv&quot;                                           
##  [4] &quot;HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1_fastqc.zip&quot;
##  [5] &quot;HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2_fastqc.zip&quot;
##  [6] &quot;HBR_Rep1Aligned.sortedByCoord.out.bam&quot;                            
##  [7] &quot;HBR_Rep1Aligned.sortedByCoord.out.bam.bai&quot;                        
##  [8] &quot;HBR_Rep1Aligned.sortedByCoord.out.distribution.txt&quot;               
##  [9] &quot;HBR_Rep1Aligned.sortedByCoord.out.featureCounts.txt&quot;              
## [10] &quot;HBR_Rep1Aligned.sortedByCoord.out.featureCounts.txt.summary&quot;      
## [11] &quot;HBR_Rep1Aligned.sortedByCoord.out.flagstat.txt&quot;                   
## [12] &quot;HBR_Rep1Aligned.sortedByCoord.out.geneBodyCoverage.curves.pdf&quot;    
## [13] &quot;HBR_Rep1Aligned.sortedByCoord.out.geneBodyCoverage.txt&quot;           
## [14] &quot;HBR_Rep1Log.final.out&quot;                                            
## [15] &quot;HBR_Rep1ReadsPerGene.out.tab&quot;                                     
## [16] &quot;run_info.json&quot;</code></pre>
</div>
<div id="qc-summary" class="section level3">
<h3><span class="header-section-number">7.1.3</span> QC Summary</h3>
<p>The tool “multiqc” can aggregate results from the multiple outputs of the pipeline and generate a single page report, which also was implemented in the <code>RcwlPipelines</code> package:</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="rna-seq-alignment-and-quantification.html#cb241-1"></a>multiqc<span class="op">$</span>dir &lt;-<span class="st"> &quot;apps/RNASeq/output&quot;</span></span>
<span id="cb241-2"><a href="rna-seq-alignment-and-quantification.html#cb241-2"></a>multiqc</span></code></pre></div>
<p>We can also run the tool using <code>Rcwl</code> locally with the option <code>docker = TRUE</code>:</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="rna-seq-alignment-and-quantification.html#cb242-1"></a><span class="kw">runCWL</span>(multiqc, <span class="dt">stderr =</span> <span class="st">&quot;&quot;</span>, <span class="dt">Args =</span> <span class="st">&quot;--preserve-entire-environment&quot;</span>, <span class="dt">docker =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div id="abundances-summary" class="section level3">
<h3><span class="header-section-number">7.1.4</span> Abundances summary</h3>
<p>The R/Bioconductor package <code>edgeR</code> can be used to calculate the RPKM and CPM abundances.</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="rna-seq-alignment-and-quantification.html#cb243-1"></a>countfiles &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;apps/RNASeq/output&quot;</span>, <span class="st">&quot;out.featureCounts.txt$&quot;</span>,</span>
<span id="cb243-2"><a href="rna-seq-alignment-and-quantification.html#cb243-2"></a>                         <span class="dt">recursive =</span> <span class="ot">TRUE</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</span>
<span id="cb243-3"><a href="rna-seq-alignment-and-quantification.html#cb243-3"></a>samples &lt;-<span class="st"> </span><span class="kw">basename</span>(<span class="kw">dirname</span>(countfiles))</span>
<span id="cb243-4"><a href="rna-seq-alignment-and-quantification.html#cb243-4"></a></span>
<span id="cb243-5"><a href="rna-seq-alignment-and-quantification.html#cb243-5"></a>rExp &lt;-<span class="st"> </span><span class="cf">function</span>(countfile){</span>
<span id="cb243-6"><a href="rna-seq-alignment-and-quantification.html#cb243-6"></a>    count1 &lt;-<span class="st"> </span><span class="kw">read.table</span>(countfile, <span class="dt">header =</span> <span class="ot">TRUE</span>)[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">7</span>)]</span>
<span id="cb243-7"><a href="rna-seq-alignment-and-quantification.html#cb243-7"></a>    rpkm1 &lt;-<span class="st"> </span>edgeR<span class="op">::</span><span class="kw">rpkm</span>(count1[,<span class="dv">3</span>,<span class="dt">drop=</span>F], <span class="dt">gene.length =</span> count1<span class="op">$</span>Length)</span>
<span id="cb243-8"><a href="rna-seq-alignment-and-quantification.html#cb243-8"></a>    cpm1 &lt;-<span class="st"> </span>edgeR<span class="op">::</span><span class="kw">cpm</span>(count1[,<span class="dv">3</span>])</span>
<span id="cb243-9"><a href="rna-seq-alignment-and-quantification.html#cb243-9"></a>    count1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(count1, rpkm1, cpm1)</span>
<span id="cb243-10"><a href="rna-seq-alignment-and-quantification.html#cb243-10"></a>    <span class="kw">colnames</span>(count1)[<span class="dv">3</span><span class="op">:</span><span class="dv">5</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;count&quot;</span>, <span class="st">&quot;rpkm&quot;</span>, <span class="st">&quot;cpm&quot;</span>)</span>
<span id="cb243-11"><a href="rna-seq-alignment-and-quantification.html#cb243-11"></a>    <span class="kw">return</span>(count1)</span>
<span id="cb243-12"><a href="rna-seq-alignment-and-quantification.html#cb243-12"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="rna-seq-alignment-and-quantification.html#cb244-1"></a><span class="kw">head</span>(<span class="kw">rExp</span>(countfiles[<span class="dv">1</span>]))</span></code></pre></div>
<pre><code>##              Geneid Length count rpkm cpm
## 1 ENSG00000223972.5   1735     0    0   0
## 2 ENSG00000227232.5   1351     0    0   0
## 3 ENSG00000278267.1     68     0    0   0
## 4 ENSG00000243485.4   1021     0    0   0
## 5 ENSG00000237613.2   1219     0    0   0
## 6 ENSG00000268020.3    840     0    0   0</code></pre>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="rna-seq-alignment-and-quantification.html#cb246-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(samples)) {</span>
<span id="cb246-2"><a href="rna-seq-alignment-and-quantification.html#cb246-2"></a>    exp1 &lt;-<span class="st"> </span><span class="kw">rExp</span>(countfiles[i])</span>
<span id="cb246-3"><a href="rna-seq-alignment-and-quantification.html#cb246-3"></a>    <span class="kw">write.table</span>(exp1, <span class="dt">file =</span> <span class="kw">paste0</span>(<span class="st">&quot;apps/RNASeq/output/&quot;</span>, samples[i],</span>
<span id="cb246-4"><a href="rna-seq-alignment-and-quantification.html#cb246-4"></a>                                    <span class="st">&quot;/&quot;</span>, samples[i], <span class="st">&quot;_abundance.tsv&quot;</span>),</span>
<span id="cb246-5"><a href="rna-seq-alignment-and-quantification.html#cb246-5"></a>                <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb246-6"><a href="rna-seq-alignment-and-quantification.html#cb246-6"></a>}</span></code></pre></div>
<p>We can also easily wrap the R function to a <code>cwlParam</code> object.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="rna-seq-alignment-and-quantification.html#cb247-1"></a>rExp &lt;-<span class="st"> </span><span class="cf">function</span>(countfile){</span>
<span id="cb247-2"><a href="rna-seq-alignment-and-quantification.html#cb247-2"></a>    count1 &lt;-<span class="st"> </span><span class="kw">read.table</span>(countfile, <span class="dt">header =</span> <span class="ot">TRUE</span>)[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">7</span>)]</span>
<span id="cb247-3"><a href="rna-seq-alignment-and-quantification.html#cb247-3"></a>    rpkm1 &lt;-<span class="st"> </span>edgeR<span class="op">::</span><span class="kw">rpkm</span>(count1[,<span class="dv">3</span>,<span class="dt">drop=</span>F], <span class="dt">gene.length =</span> count1<span class="op">$</span>Length)</span>
<span id="cb247-4"><a href="rna-seq-alignment-and-quantification.html#cb247-4"></a>    cpm1 &lt;-<span class="st"> </span>edgeR<span class="op">::</span><span class="kw">cpm</span>(count1[,<span class="dv">3</span>])</span>
<span id="cb247-5"><a href="rna-seq-alignment-and-quantification.html#cb247-5"></a>    count1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(count1, rpkm1, cpm1)</span>
<span id="cb247-6"><a href="rna-seq-alignment-and-quantification.html#cb247-6"></a>    <span class="kw">colnames</span>(count1)[<span class="dv">3</span><span class="op">:</span><span class="dv">5</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;count&quot;</span>, <span class="st">&quot;rpkm&quot;</span>, <span class="st">&quot;cpm&quot;</span>)</span>
<span id="cb247-7"><a href="rna-seq-alignment-and-quantification.html#cb247-7"></a>    <span class="kw">write.table</span>(exp1, <span class="dt">file =</span> <span class="kw">paste0</span>(tools<span class="op">::</span><span class="kw">file_path_sans_ext</span>(<span class="kw">basename</span>(countfile)),</span>
<span id="cb247-8"><a href="rna-seq-alignment-and-quantification.html#cb247-8"></a>                                    <span class="st">&quot;_abundance.tsv&quot;</span>),</span>
<span id="cb247-9"><a href="rna-seq-alignment-and-quantification.html#cb247-9"></a>                <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb247-10"><a href="rna-seq-alignment-and-quantification.html#cb247-10"></a>}</span>
<span id="cb247-11"><a href="rna-seq-alignment-and-quantification.html#cb247-11"></a>p1 &lt;-<span class="st"> </span><span class="kw">InputParam</span>(<span class="dt">id =</span> <span class="st">&quot;countfile&quot;</span>, <span class="dt">type =</span> <span class="st">&quot;File&quot;</span>,</span>
<span id="cb247-12"><a href="rna-seq-alignment-and-quantification.html#cb247-12"></a>                 <span class="dt">prefix =</span> <span class="st">&quot;countfile=&quot;</span>, <span class="dt">separate =</span> <span class="ot">FALSE</span>)</span>
<span id="cb247-13"><a href="rna-seq-alignment-and-quantification.html#cb247-13"></a>o1 &lt;-<span class="st"> </span><span class="kw">OutputParam</span>(<span class="dt">id =</span> <span class="st">&quot;exp&quot;</span>, <span class="dt">type =</span> <span class="st">&quot;File&quot;</span>, <span class="dt">glob =</span> <span class="st">&quot;*abundance.tsv&quot;</span>)</span>
<span id="cb247-14"><a href="rna-seq-alignment-and-quantification.html#cb247-14"></a>rExpr &lt;-<span class="st"> </span><span class="kw">cwlParam</span>(<span class="dt">baseCommand =</span> rExp,</span>
<span id="cb247-15"><a href="rna-seq-alignment-and-quantification.html#cb247-15"></a>                  <span class="dt">inputs =</span> <span class="kw">InputParamList</span>(p1),</span>
<span id="cb247-16"><a href="rna-seq-alignment-and-quantification.html#cb247-16"></a>                  <span class="dt">outputs =</span> <span class="kw">OutputParamList</span>(o1))</span></code></pre></div>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="rna-seq-alignment-and-quantification.html#cb248-1"></a>countfiles &lt;-<span class="st"> </span><span class="kw">as.list</span>(countfiles)</span>
<span id="cb248-2"><a href="rna-seq-alignment-and-quantification.html#cb248-2"></a><span class="kw">names</span>(countfiles) &lt;-<span class="st"> </span>samples</span>
<span id="cb248-3"><a href="rna-seq-alignment-and-quantification.html#cb248-3"></a>inputList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">countfile =</span> countfiles)</span>
<span id="cb248-4"><a href="rna-seq-alignment-and-quantification.html#cb248-4"></a></span>
<span id="cb248-5"><a href="rna-seq-alignment-and-quantification.html#cb248-5"></a>res1 &lt;-<span class="st"> </span><span class="kw">runCWLBatch</span>(rExpr, <span class="dt">outdir =</span> <span class="st">&quot;apps/RNASeq/output&quot;</span>, inputList,</span>
<span id="cb248-6"><a href="rna-seq-alignment-and-quantification.html#cb248-6"></a>                    <span class="dt">BPPARAM =</span> <span class="kw">BatchtoolsParam</span>(<span class="dt">workers =</span> <span class="kw">length</span>(samples),</span>
<span id="cb248-7"><a href="rna-seq-alignment-and-quantification.html#cb248-7"></a>                                              <span class="dt">cluster =</span> <span class="st">&quot;sge&quot;</span>,</span>
<span id="cb248-8"><a href="rna-seq-alignment-and-quantification.html#cb248-8"></a>                                              <span class="dt">template =</span> <span class="st">&quot;apps/data/sge.tmpl&quot;</span>,</span>
<span id="cb248-9"><a href="rna-seq-alignment-and-quantification.html#cb248-9"></a>                                              <span class="dt">resources =</span> <span class="kw">list</span>(<span class="dt">threads =</span> <span class="dv">16</span>,</span>
<span id="cb248-10"><a href="rna-seq-alignment-and-quantification.html#cb248-10"></a>                                                               <span class="dt">queue =</span> <span class="st">&quot;all.q&quot;</span>,</span>
<span id="cb248-11"><a href="rna-seq-alignment-and-quantification.html#cb248-11"></a>                                                               <span class="dt">jobname =</span> <span class="st">&quot;Exp&quot;</span>),</span>
<span id="cb248-12"><a href="rna-seq-alignment-and-quantification.html#cb248-12"></a>                                              <span class="dt">log =</span> <span class="ot">TRUE</span>, <span class="dt">logdir =</span> <span class="st">&quot;.&quot;</span>, <span class="dt">progressbar =</span> <span class="ot">TRUE</span>),</span>
<span id="cb248-13"><a href="rna-seq-alignment-and-quantification.html#cb248-13"></a>                    <span class="dt">stderr =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="transcriptome-quantification" class="section level2">
<h2><span class="header-section-number">7.2</span> transcriptome quantification</h2>
<p>There are many tools available for transcriptome quantification, such as kallisto, StringTie, salmon, Trinity and so on.</p>
<div id="kallisto" class="section level3">
<h3><span class="header-section-number">7.2.1</span> Kallisto</h3>
<p>The kallisto is a tool for quantifying abundances of transcripts with raw fastq reads and indexed reference transcriptomes. The <code>kallisto_index</code> is used to build an index file from reference transcriptome fasta.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="rna-seq-alignment-and-quantification.html#cb249-1"></a>kallisto_index<span class="op">$</span>fasta &lt;-<span class="st"> &quot;apps/data/gencode.v25.transcripts.fa&quot;</span></span>
<span id="cb249-2"><a href="rna-seq-alignment-and-quantification.html#cb249-2"></a>kallisto_index<span class="op">$</span>index &lt;-<span class="st"> &quot;gencode.v25.transcripts_kallisto&quot;</span></span>
<span id="cb249-3"><a href="rna-seq-alignment-and-quantification.html#cb249-3"></a><span class="kw">runCWL</span>(kallisto_index, <span class="dt">outdir =</span> <span class="st">&quot;apps/data/&quot;</span>, <span class="dt">stderr =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>The <code>kallisto_quant</code> runs the quantification algorithm to estimate the transcripts expression abundances.</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="rna-seq-alignment-and-quantification.html#cb250-1"></a>inputList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">fastq =</span> files)</span>
<span id="cb250-2"><a href="rna-seq-alignment-and-quantification.html#cb250-2"></a>paramList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">index =</span> <span class="st">&quot;apps/data/gencode.v25.transcripts_kallisto&quot;</span>,</span>
<span id="cb250-3"><a href="rna-seq-alignment-and-quantification.html#cb250-3"></a>                  <span class="dt">threads =</span> <span class="dv">16</span>)</span>
<span id="cb250-4"><a href="rna-seq-alignment-and-quantification.html#cb250-4"></a></span>
<span id="cb250-5"><a href="rna-seq-alignment-and-quantification.html#cb250-5"></a>res2 &lt;-<span class="st"> </span><span class="kw">runCWLBatch</span>(kallisto_quant, <span class="dt">outdir =</span> <span class="st">&quot;apps/RNASeq/output&quot;</span>, inputList, paramList,</span>
<span id="cb250-6"><a href="rna-seq-alignment-and-quantification.html#cb250-6"></a>                    <span class="dt">BPPARAM =</span> <span class="kw">BatchtoolsParam</span>(<span class="dt">workers =</span> <span class="kw">length</span>(samples),</span>
<span id="cb250-7"><a href="rna-seq-alignment-and-quantification.html#cb250-7"></a>                                              <span class="dt">cluster =</span> <span class="st">&quot;sge&quot;</span>,</span>
<span id="cb250-8"><a href="rna-seq-alignment-and-quantification.html#cb250-8"></a>                                              <span class="dt">template =</span> <span class="st">&quot;apps/data/sge.tmpl&quot;</span>,</span>
<span id="cb250-9"><a href="rna-seq-alignment-and-quantification.html#cb250-9"></a>                                              <span class="dt">resources =</span> <span class="kw">list</span>(<span class="dt">threads =</span> <span class="dv">16</span>,</span>
<span id="cb250-10"><a href="rna-seq-alignment-and-quantification.html#cb250-10"></a>                                                               <span class="dt">queue =</span> <span class="st">&quot;all.q&quot;</span>,</span>
<span id="cb250-11"><a href="rna-seq-alignment-and-quantification.html#cb250-11"></a>                                                               <span class="dt">jobname =</span> <span class="st">&quot;kallisto&quot;</span>),</span>
<span id="cb250-12"><a href="rna-seq-alignment-and-quantification.html#cb250-12"></a>                                              <span class="dt">log =</span> <span class="ot">TRUE</span>, <span class="dt">logdir =</span> <span class="st">&quot;.&quot;</span>, <span class="dt">progressbar =</span> <span class="ot">TRUE</span>),</span>
<span id="cb250-13"><a href="rna-seq-alignment-and-quantification.html#cb250-13"></a>                    <span class="dt">stderr =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>Here are the results:</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="rna-seq-alignment-and-quantification.html#cb251-1"></a><span class="kw">list.files</span>(<span class="st">&quot;apps/RNASeq/output/HBR_Rep1&quot;</span>, <span class="st">&quot;abundance&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;abundance.h5&quot;           &quot;abundance.tsv&quot;         
## [3] &quot;HBR_Rep1_abundance.tsv&quot;</code></pre>
</div>
</div>
<div id="single-cell-rnaseq" class="section level2">
<h2><span class="header-section-number">7.3</span> single cell RNAseq</h2>
</div>
<div id="mirna" class="section level2">
<h2><span class="header-section-number">7.4</span> miRNA</h2>
<p>The <code>miRDeep2</code> is one of the popular tools for discovering known and novel miRNAs from small RNA sequencing data. We wrapped the mapping and quantification steps into <code>miRDeep2PL</code>. Here is the instructions for <code>miRDeep2</code>: <a href="https://github.com/rajewsky-lab/mirdeep2" class="uri">https://github.com/rajewsky-lab/mirdeep2</a>.</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="rna-seq-alignment-and-quantification.html#cb253-1"></a><span class="kw">plotCWL</span>(miRDeep2PL)</span></code></pre></div>
<div id="htmlwidget-2688ed0a1907c4ecf1e7" style="width:672px;height:480px;" class="DiagrammeR html-widget"></div>
<script type="application/json" data-for="htmlwidget-2688ed0a1907c4ecf1e7">{"x":{"diagram":"graph TB\nA(reads)-->R{Mapper}\nB(format)-->R{Mapper}\nC(adapter)-->R{Mapper}\nE(genome)-->R{Mapper}\nR{Mapper}-->P((preads))\nR{Mapper}-->Q((arf))\nP((preads))-->S{miRDeep2}\nE(genome)-->S{miRDeep2}\nQ((arf))-->S{miRDeep2}\nF(miRef)-->S{miRDeep2}\nG(miOther)-->S{miRDeep2}\nH(precursors)-->S{miRDeep2}\nI(species)-->S{miRDeep2}\nS{miRDeep2}-->J((csvfiles))\nS{miRDeep2}-->K((htmls))\nS{miRDeep2}-->L((bed))\nS{miRDeep2}-->M((expression))\nS{miRDeep2}-->N((mirna_results))\nS{miRDeep2}-->O((pdfs))"},"evals":[],"jsHooks":[]}</script>
<p>We use the data from <code>mirdeep2</code> github repository as an example.
<a href="https://github.com/rajewsky-lab/mirdeep2/tree/master/tutorial_dir" class="uri">https://github.com/rajewsky-lab/mirdeep2/tree/master/tutorial_dir</a></p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="rna-seq-alignment-and-quantification.html#cb254-1"></a><span class="kw">list.files</span>(<span class="st">&quot;apps/miRNA/tutorial_dir&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;cel_cluster.fa&quot;                 &quot;mature_ref_other_species.fa&quot;   
## [3] &quot;mature_ref_this_species.fa&quot;     &quot;precursors_ref_this_species.fa&quot;
## [5] &quot;reads_collapsed.fa&quot;             &quot;reads.fa&quot;                      
## [7] &quot;run_tut.sh&quot;                     &quot;sample_result.html&quot;</code></pre>
<div id="prepare-reference" class="section level3">
<h3><span class="header-section-number">7.4.1</span> Prepare reference</h3>
<p>First, we need to build indexes for the miRNA reference with <code>bowtie-build</code>. This is only required to be performed once for each refernce genome.</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="rna-seq-alignment-and-quantification.html#cb256-1"></a>bowtie_build<span class="op">$</span>ref &lt;-<span class="st"> &quot;apps/miRNA/tutorial_dir/cel_cluster.fa&quot;</span></span>
<span id="cb256-2"><a href="rna-seq-alignment-and-quantification.html#cb256-2"></a>bowtie_build<span class="op">$</span>outPrefix &lt;-<span class="st"> &quot;cel_cluster&quot;</span></span>
<span id="cb256-3"><a href="rna-seq-alignment-and-quantification.html#cb256-3"></a>idxRes &lt;-<span class="st"> </span><span class="kw">runCWL</span>(bowtie_build, <span class="dt">outdir =</span> <span class="st">&quot;apps/miRNA/output/genome&quot;</span>, <span class="dt">stderr =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb256-4"><a href="rna-seq-alignment-and-quantification.html#cb256-4"></a><span class="kw">file.copy</span>(<span class="st">&quot;apps/miRNA/tutorial_dir/cel_cluster.fa&quot;</span>, <span class="st">&quot;apps/miRNA/output/genome/cel_cluster.fa&quot;</span>)</span></code></pre></div>
<p>Here are the indexed reference files.</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="rna-seq-alignment-and-quantification.html#cb257-1"></a><span class="kw">dir_info</span>(<span class="st">&quot;apps/miRNA/output/genome&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(path, size)</span></code></pre></div>
<pre><code>## # A tibble: 7 x 2
##   path                                                   size
##   &lt;fs::path&gt;                                      &lt;fs::bytes&gt;
## 1 apps/miRNA/output/genome/cel_cluster.1.ebwt              4M
## 2 apps/miRNA/output/genome/cel_cluster.2.ebwt             812
## 3 apps/miRNA/output/genome/cel_cluster.3.ebwt              26
## 4 apps/miRNA/output/genome/cel_cluster.4.ebwt           1.57K
## 5 apps/miRNA/output/genome/cel_cluster.fa               6.47K
## 6 apps/miRNA/output/genome/cel_cluster.rev.1.ebwt          4M
## 7 apps/miRNA/output/genome/cel_cluster.rev.2.ebwt         812</code></pre>
</div>
<div id="run-mirdeep2-pipeline" class="section level3">
<h3><span class="header-section-number">7.4.2</span> Run miRDeep2 pipeline</h3>
<p>First we need to prepare the inputs into ‘inputList’ and ‘paramList’ for parallel execution.</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="rna-seq-alignment-and-quantification.html#cb259-1"></a><span class="kw">inputs</span>(miRDeep2PL)</span></code></pre></div>
<pre><code>## inputs:
##   reads (File):  
##   format (string):  -c
##   adapter (string):  
##   len (int):  18
##   genome (File):  
##   miRef  ( File|string ): 
##   miOther  ( File|string ): 
##   precursors  ( File|string ): 
##   species (string):</code></pre>
<p>To mimic multiple samples analysis, here we just repeat to use the input reads.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="rna-seq-alignment-and-quantification.html#cb261-1"></a>reads &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">sample1 =</span> <span class="st">&quot;apps/miRNA/tutorial_dir/reads.fa&quot;</span>,</span>
<span id="cb261-2"><a href="rna-seq-alignment-and-quantification.html#cb261-2"></a>              <span class="dt">sample2 =</span> <span class="st">&quot;apps/miRNA/tutorial_dir/reads.fa&quot;</span>)</span>
<span id="cb261-3"><a href="rna-seq-alignment-and-quantification.html#cb261-3"></a></span>
<span id="cb261-4"><a href="rna-seq-alignment-and-quantification.html#cb261-4"></a>inputList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">reads =</span> reads)</span>
<span id="cb261-5"><a href="rna-seq-alignment-and-quantification.html#cb261-5"></a>paramList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">adapter =</span> <span class="st">&quot;TCGTATGCCGTCTTCTGCTTGT&quot;</span>,</span>
<span id="cb261-6"><a href="rna-seq-alignment-and-quantification.html#cb261-6"></a>                  <span class="dt">genome =</span> <span class="st">&quot;apps/miRNA/output/genome/cel_cluster.fa&quot;</span>,</span>
<span id="cb261-7"><a href="rna-seq-alignment-and-quantification.html#cb261-7"></a>                  <span class="dt">miRef =</span> <span class="st">&quot;apps/miRNA/tutorial_dir/mature_ref_this_species.fa&quot;</span>,</span>
<span id="cb261-8"><a href="rna-seq-alignment-and-quantification.html#cb261-8"></a>                  <span class="dt">miOther =</span> <span class="st">&quot;apps/miRNA/tutorial_dir/mature_ref_other_species.fa&quot;</span>,</span>
<span id="cb261-9"><a href="rna-seq-alignment-and-quantification.html#cb261-9"></a>                  <span class="dt">precursors =</span> <span class="st">&quot;apps/miRNA/tutorial_dir/precursors_ref_this_species.fa&quot;</span>,</span>
<span id="cb261-10"><a href="rna-seq-alignment-and-quantification.html#cb261-10"></a>                  <span class="dt">species =</span> <span class="st">&quot;C.elegans&quot;</span>)</span></code></pre></div>
<p>Let’s run the pipeline with two computing nodes.</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="rna-seq-alignment-and-quantification.html#cb262-1"></a>mirRes &lt;-<span class="st"> </span><span class="kw">runCWLBatch</span>(miRDeep2PL, <span class="dt">outdir =</span> <span class="st">&quot;apps/miRNA/output/&quot;</span>, inputList, paramList,</span>
<span id="cb262-2"><a href="rna-seq-alignment-and-quantification.html#cb262-2"></a>                      <span class="dt">BPPARAM =</span> <span class="kw">BatchtoolsParam</span>(<span class="dt">workers =</span> <span class="dv">2</span>, <span class="dt">cluster =</span> <span class="st">&quot;sge&quot;</span>,</span>
<span id="cb262-3"><a href="rna-seq-alignment-and-quantification.html#cb262-3"></a>                                                <span class="dt">template =</span> <span class="st">&quot;apps/data/sge.tmpl&quot;</span>,</span>
<span id="cb262-4"><a href="rna-seq-alignment-and-quantification.html#cb262-4"></a>                                                <span class="dt">resources =</span> <span class="kw">list</span>(<span class="dt">jobname =</span> <span class="st">&quot;miRNA&quot;</span>,</span>
<span id="cb262-5"><a href="rna-seq-alignment-and-quantification.html#cb262-5"></a>                                                                 <span class="dt">queue =</span> <span class="st">&quot;all.q&quot;</span>,</span>
<span id="cb262-6"><a href="rna-seq-alignment-and-quantification.html#cb262-6"></a>                                                                 <span class="dt">threads =</span> <span class="dv">2</span>),</span>
<span id="cb262-7"><a href="rna-seq-alignment-and-quantification.html#cb262-7"></a>                                                <span class="dt">log=</span><span class="ot">TRUE</span>, <span class="dt">logdir=</span><span class="st">&quot;.&quot;</span>, <span class="dt">progressbar =</span> T),</span>
<span id="cb262-8"><a href="rna-seq-alignment-and-quantification.html#cb262-8"></a>                      <span class="dt">stderr =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>Here are the results:</p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="rna-seq-alignment-and-quantification.html#cb263-1"></a><span class="kw">dir_info</span>(<span class="st">&quot;apps/miRNA/output/sample1&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(path, size)</span></code></pre></div>
<pre><code>## # A tibble: 10 x 2
##    path                                                                size
##    &lt;fs::path&gt;                                                    &lt;fs::byte&gt;
##  1 apps/miRNA/output/sample1/expression_26_09_2019_t_20_10_36.h…     21.15K
##  2 apps/miRNA/output/sample1/expression_analyses                         4K
##  3 apps/miRNA/output/sample1/miRNAs_expressed_all_samples_26_09…        441
##  4 apps/miRNA/output/sample1/mirna_results_26_09_2019_t_20_10_36         4K
##  5 apps/miRNA/output/sample1/pdfs_26_09_2019_t_20_10_36                  4K
##  6 apps/miRNA/output/sample1/reads_collapsed.arf                     61.36K
##  7 apps/miRNA/output/sample1/reads_collapsed.fa                      59.31K
##  8 apps/miRNA/output/sample1/result_26_09_2019_t_20_10_36.bed           747
##  9 apps/miRNA/output/sample1/result_26_09_2019_t_20_10_36.csv         4.39K
## 10 apps/miRNA/output/sample1/result_26_09_2019_t_20_10_36.html       42.73K</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dna-seq-variant-calling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="neoantigen-prediction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/07-RNASeq.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["Rcwl-Pipelines.pdf", "Rcwl-Pipelines.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
